File: app/src/main/java/com/example/student/bricklist/ProjectAdapter.java
package com.example.student.bricklist;

import android.content.Context;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.Adapter;
import android.widget.BaseAdapter;
import android.widget.ImageView;
import android.widget.TextView;
import android.widget.Toast;

import java.util.List;

/**
 * Created by student on 28.05.2018.
 */

public class ProjectAdapter extends BaseAdapter {

    List<Project> projectList;
    Context context;

    public ProjectAdapter(Context context, List<Project> projectList){
        this.projectList = projectList;
        this.context = context;
    }

    public void changeDataSet(List<Project> projectList){
        this.projectList = projectList;
        notifyDataSetChanged();
    }

    public void addNewProject(Project project){
        this.projectList.add(project);
        notifyDataSetChanged();
    }

    @Override
    public int getCount() {
        return projectList.size();
    }

    @Override
    public Project getItem(int i) {
        return projectList.get(i);
    }

    @Override
    public long getItemId(int i) {
        return projectList.get(i).getId();
    }

    @Override
    public View getView(final int i, View view, ViewGroup viewGroup) {
        View row = view;
        if(view == null){
            LayoutInflater vi;
            vi = LayoutInflater.from(context);
            row = vi.inflate(R.layout.project_row_layout, null);
        }

        TextView projectName = row.findViewById(R.id.projectName);
        ImageView deleteProject = row.findViewById(R.id.deleteProject);
        ImageView archiveProject = row.findViewById(R.id.archiveProject);

        final Project project = projectList.get(i);

        projectName.setText(project.getName());

        deleteProject.setOnClickListener(new View.OnClickListener(){
            public void onClick(View v) {
                projectList.remove(project);
                notifyDataSetChanged();
            }
        });

        archiveProject.setOnClickListener(new View.OnClickListener(){
            public void onClick(View v) {
                Toast.makeText(context, "Project archived", Toast.LENGTH_LONG).show();
            }
        });

        projectName.setOnClickListener(new View.OnClickListener(){
            public void onClick(View v) {
                Toast.makeText(context, "Project opened", Toast.LENGTH_LONG).show();
            }
        });

        return row;
    }
}

File: app/src/main/java/com/piciu/bricklist/ProjectAdapter.kt
package com.piciu.bricklist

import android.content.Context
import android.content.Intent
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.BaseAdapter
import android.widget.ImageView
import android.widget.TextView
import android.widget.Toast

class ProjectAdapter(private val context: Context) : BaseAdapter() {

    private val databaseHelper: DataBaseHelper = DataBaseHelper(context)

    val projectList: ArrayList<Project> = databaseHelper.getProjects()

    fun addNewProject(project: Project) {
        this.projectList.add(project)
        notifyDataSetChanged()

        databaseHelper.addProject(project)
    }

    override fun getCount(): Int {
        return projectList.size
    }

    override fun getItem(i: Int): Project {
        return projectList[i]
    }

    override fun getItemId(i: Int): Long {
        return i.toLong()
    }

    override fun getView(i: Int, view: View?, viewGroup: ViewGroup): View {
        var row = view
        if (row == null) {
            val inflater: LayoutInflater = context.getSystemService(Context.LAYOUT_INFLATER_SERVICE) as LayoutInflater
            row = inflater.inflate(R.layout.project_row_layout, viewGroup, false)
        }

        val projectName = row!!.findViewById<TextView>(R.id.projectName)
        val deleteProject = row.findViewById<ImageView>(R.id.deleteProject)
        val archiveProject = row.findViewById<ImageView>(R.id.archiveProject)

        val project = projectList[i]

        projectName.text = project.name

        deleteProject.setOnClickListener {
            projectList.remove(project)
            notifyDataSetChanged()

            databaseHelper.deleteProject(project)
        }

        archiveProject.setOnClickListener {
            Toast.makeText(context, "Project archived", Toast.LENGTH_LONG).show()

            project.archived = true
            databaseHelper.changeProjectValues(project)
        }

        projectName.setOnClickListener {
            ProjectActivity.project = project
            val intent = Intent(context,ProjectActivity::class.java)
            context.startActivity(intent)
        }

        return row
    }
}
